megamek/src/megamek/client/ui/swing/OffBoardTargetOverlay.java

```diff
@@ -0,0 +1,365 @@
+import megamek.common.IGame;
+import megamek.common.actions.ArtilleryAttackAction;
+import megamek.common.actions.WeaponAttackAction;
+import megamek.common.util.ImageUtil;
+import megamek.common.util.MegaMekFile;
+import megamek.common.IPlayer;
+import megamek.common.Mounted;
+import megamek.common.OffBoardDirection;
+import megamek.common.Targetable;
+
+/**
+ * This class handles the display and logic for the off board targeting overlay.
+ *
+ */
+public class OffBoardTargetOverlay implements IDisplayable {
+    private static final int EDGE_OFFSET = 5;
+    private static final int WIDE_EDGE_SIZE = 60;
+    private static final int NARROW_EDGE_SIZE = 40;
+    private static final String FILENAME_OFFBOARD_TARGET_IMAGE = "OffBoardTarget.png";
+    
+    private boolean isHit = false;
+    private ClientGUI clientgui;
+    private Map<OffBoardDirection, Rectangle> buttons = new HashMap<>();
+    private TargetingPhaseDisplay targetingPhaseDisplay;
+    private Image offBoardTargetImage;
+    
+    private IGame getCurrentGame() {
+        return clientgui.getClient().getGame();
+    }
+    
+    private IPlayer getCurrentPlayer() {
+        return clientgui.getClient().getLocalPlayer();
+    }
+    
+    /**
+     * Sets a reference to a TargetingPhaseDisplay. Used to communicate a generated attack to it.
+     */
+    public void setTargetingPhaseDisplay(TargetingPhaseDisplay tpd) {
+        targetingPhaseDisplay = tpd;
+    }    
+    
+    public OffBoardTargetOverlay(ClientGUI clientgui) {
+        this.clientgui = clientgui;
+        
+        offBoardTargetImage = ImageUtil.loadImageFromFile(
+                new MegaMekFile(Configuration.miscImagesDir(), FILENAME_OFFBOARD_TARGET_IMAGE)
+                        .toString());
+        
+        //Maybe TODO: display dimmed version of off-board icon during movement phase OR targeting phase when weapon is ineligible to fire 
+        //Maybe TODO: maybe tooltips?
+    }
+    
+    /**
+     * Logic that determines if this overlay should be visible.
+     */
+    private boolean shouldBeVisible() {
+        // only relevant if it's our turn in the targeting phase
+        boolean visible = clientgui.getClient().isMyTurn() &&
+                (getCurrentGame().getPhase() == IGame.Phase.PHASE_TARGETING);
+        
+        if(!visible) {
+            return false;
+        }
+        
+        Mounted selectedArtilleryWeapon = clientgui.getBoardView().getSelectedArtilleryWeapon();
+        
+        // only relevant if we've got an artillery weapon selected for one of our own units
+        if(selectedArtilleryWeapon == null ) {
+            return false;
+        }
+        
+        // the artillery weapon needs to be using non-homing ammo
+        Mounted ammo = selectedArtilleryWeapon.getLinked();
+        AmmoType ammoType = (AmmoType) ammo.getType();
+        if((ammoType.getMunitionType() == AmmoType.M_HOMING) 
+                && ammo.curMode().equals("Homing")) {
```

Comment:
Do we have constants/statics for these modes?

Reply:
Not at this time. There are quite a few places where this comparison is made, so I think I'm going to hold off on it for this PR, for the sake of minimizing its impact.